src/ssh/sftp_client.cpp

```diff
@@ -161,6 +157,10 @@ bool SFTPClient::push_dir(const fs::path& source_path, const fs::path& target_pa
     auto local_iter = MP_FILEOPS.recursive_dir_iterator(source_path, err);
     if (err)
         throw SFTPError{"cannot open local directory {}: {}", source_path, err.message()};
+
+    std::vector<std::pair<fs::path, fs::perms>> subdirectory_perms{
+        {target_path, MP_FILEOPS.status(source_path, err).permissions()}};
```

Comment:
This type is repeated, could you add a `typedef`?

Reply:
It's not really repeated. One vector has `std::pair<fs::path, fs::perms>` and the other has `std::pair<fs::path, mode_t>`, since one will be used by `std::filesystem::permissions` and the other will be used by `sftp_chmod` :smile:

Reply:
Oh, right, didn't catch that. Thanks!